mlfinlab/data_structures/imbalance_data_structures.py

```diff
@@ -0,0 +1,287 @@
+"""
+Advances in Financial Machine Learning, Marcos Lopez de Prado
+Chapter 2: Financial Data Structures
+
+This module contains the functions to help users create structured financial data from raw unstructured data,
+in the form of tick, volume, and dollar imbalance bars.
+
+These bars are used throughout the text book (Advances in Financial Machine Learning, By Marcos Lopez de Prado, 2018, pg 25)
+to build the more interesting features for predicting financial time series data.
+
+These financial data structures have better statistical properties when compared to those based on fixed time interval sampling.
+A great paper to read more about this is titled: The Volume Clock: Insights into the high frequency paradigm, Lopez de Prado, et al
+"""
+
+# Imports
+import pandas as pd
+import numpy as np
+from mlfinlab.data_structures.fast_ewma import ewma
+
+
+def _update_counters(cache, flag, exp_num_ticks_init):
```

Comment:
It's better to name function by their straight use. E.g., you update your values and return them, so the name `get_updated_counters` would be better. `update_counters` is ok if you reassign variables in, e.g., class method